memo_wise.gemspec

```diff
@@ -1,13 +1,27 @@
 # frozen_string_literal: true
 
-require_relative "lib/memo_wise/version"
-
 Gem::Specification.new do |spec|
-  spec.name     = "memo_wise"
-  spec.version  = MemoWise::VERSION
-  spec.summary  = "The wise choice for Ruby memoization"
+  spec.name = "memo_wise"
+
+  # After loading the `VERSION` constant, save it to a variable to use in this
+  # gemspec and then undefine it. This allows benchmarks to compare multiple
+  # versions of MemoWise against each other without inadvertently sharing the
+  # same `VERSION` constant.
+  # NOTE: It's important that we use `load` instead of `require_relative`
+  # because the latter only loads a file once, and so by undefining the
+  # `VERSION` constant we can make it difficult to access that value again
+  # later. For more context, see: https://github.com/panorama-ed/memo_wise/pull/370#issuecomment-2560268423
+  spec.version = Module.new.tap do |mod|
+    # NOTE: We fully qualify `Kernel` here because Rubygems defines its own
+    # `load` method, which is used by default when this code is executed by
+    # `gem build memo_wise.gemspec`. That `load` method does not support the
+    # optional "wrap" parameter we pass through as `mod`. For more context, see:
+    # https://github.com/simplecov-ruby/simplecov/issues/557#issuecomment-2630782358
+    Kernel.load("lib/memo_wise/version.rb", mod)
+  end::MemoWise::VERSION
```

Comment:
This is so cool. It's working right? :heart_decoration: I'm going to switch to use this pattern everywhere!

Reply:
Yep, works for me!

Reply:
Also note [these comments](https://github.com/panorama-ed/memo_wise/compare/14bb4355e6fe96c7285d4ca643022acfbcd2eb9d..403a4e75d5ef9806e1bf7daa4c33fc0788365765) I just added